src/test/java/uk/gov/ida/eventemitter/EventEmitterIntegrationTest.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.ida.eventemitter;
+
+import cloud.localstack.LocalstackTestRunner;
+import cloud.localstack.TestUtils;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.CreateQueueRequest;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+import org.joda.time.DateTime;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+@RunWith(LocalstackTestRunner.class)
+public class EventEmitterIntegrationTest {
+
+    private static final AmazonSQS CLIENT_SQS = TestUtils.getClientSQS();
+    private static final CreateQueueRequest CREATE_QUEUE_REQUEST = new CreateQueueRequest("MyQueue");
+    private static final String KEY = "aesEncryptionKey";
+    private static final String INIT_VECTOR = "encryptionIntVec";
+    private static String queueUrl;
+
+    private EventEmitter eventEmitter;
+
+    @Before
+    public void setUp() {
+        final Injector injector = Guice.createInjector(new AbstractModule() {
+            @Override
+            protected void configure() {
+            }
+
+            @Provides
+            @Singleton
+            private AmazonSQS getAmazonSqs() {
+                return CLIENT_SQS;
+            }
+
+            @Provides
+            @Singleton
+            @Named("EventQueueUrl")
+            public String getEventEmitter(AmazonSQS sqs) {
+                queueUrl = sqs.createQueue(CREATE_QUEUE_REQUEST).getQueueUrl();
+                return queueUrl;
+            }
+
+            @Provides
+            @Singleton
+            private SqsClient getSqsClient(AmazonSQS sqs, @Named("EventQueueUrl") String eventQueueUrl) {
+                return new SqsClient(sqs, eventQueueUrl);
+            }
+
+            @Provides
+            @Singleton
+            private LocalEncrypter getEncrypter() {
```

Comment:
Should return just an `Encrypter` rather than a `LocalEncrypter`?

Reply:
Changing from LocalEncrypter to Encrypter breaks the integration test.

Reply:
That's interesting! We've got the interface there, and the eventEmitter only needs the interface.

Conceptually, I think either we should be able to aways pass around the interface, or we should not have the interface at all, and just always pass around the concrete implementation.

Reply:
Fixed.